cli/src/config.rs

```diff
@@ -276,30 +282,37 @@ impl ConfigEnv {
             home_dir: home_dir.clone(),
             jj_config: env::var("JJ_CONFIG").ok(),
         };
-        Ok(ConfigEnv {
+        ConfigEnv {
             home_dir,
             repo_path: None,
-            user_config_path: env.resolve()?,
+            user_config_paths: env.resolve(),
             repo_config_path: None,
             command: None,
-        })
+        }
     }
 
     pub fn set_command_name(&mut self, command: String) {
         self.command = Some(command);
     }
 
-    /// Returns a path to the user-specific config file or directory.
-    pub fn user_config_path(&self) -> Option<&Path> {
-        self.user_config_path.as_ref().map(ConfigPath::as_path)
+    /// Returns the paths to the user-specific config files or directories.
+    pub fn user_config_paths(&self) -> Vec<&Path> {
```

Comment:
nit: maybe better to return `impl Iterator<Item = &Path>` because this function is mainly used internally.